misp_modules/modules/expansion/virustotal.py

```diff
@@ -9,217 +9,218 @@
-        for feature_type, relationship in feature_types.items():
-            for feature in ('undetected_{}_samples', 'detected_{}_samples'):
-                for sample in req.get(feature.format(feature_type), [])[:self.limit]:
-                    status_code = self.parse_hash(sample[hash_type], False, uuid, relationship)
-                    if status_code != 200:
-                        return status_code
-        if req.get(whois):
-            whois_object = MISPObject(whois)
-            whois_object.add_attribute('text', type='text', value=req[whois])
+    def parse_domain(self, domain: str) -> str:
+        domain_report = self.client.get_object(f'/domains/{domain}')
+
+        # DOMAIN
+        domain_object = self.create_misp_object(domain_report)
+
+        # WHOIS
+        if domain_report.whois:
+            whois_object = MISPObject('whois')
+            whois_object.add_attribute('text', type='text', value=domain_report.whois)
             self.misp_event.add_object(**whois_object)
-        return self.parse_related_urls(req, recurse, uuid)
-
-    def parse_hash(self, sample, recurse=False, uuid=None, relationship=None):
-        req = requests.get(self.base_url.format('file'), params={'apikey': self.apikey, 'resource': sample}, proxies=self.proxies)
-        status_code = req.status_code
-        if req.status_code == 200:
-            req = req.json()
-            vt_uuid = self.parse_vt_object(req)
-            file_attributes = []
-            for hash_type in ('md5', 'sha1', 'sha256'):
-                if req.get(hash_type):
-                    file_attributes.append({'type': hash_type, 'object_relation': hash_type,
-                                            'value': req[hash_type]})
-            if file_attributes:
-                file_object = MISPObject('file')
-                for attribute in file_attributes:
-                    file_object.add_attribute(**attribute)
-                file_object.add_reference(vt_uuid, 'analyzed-with')
-                if uuid and relationship:
-                    file_object.add_reference(uuid, relationship)
+
+        # SIBLINGS
+        siblings_iterator = self.client.iterator(f'/domains/{domain_report.id}/siblings', limit=self.limit)
+        for sibling in siblings_iterator:
+            attr = MISPAttribute()
+            attr.from_dict(**dict(type='domain', value=sibling.id))
+            self.misp_event.add_attribute(**attr)
+            domain_object.add_reference(attr.uuid, 'sibling-of')
+
+        # SUBDOMAINS
+        subdomains_iterator = self.client.iterator(f'/domains/{domain_report.id}/subdomains', limit=self.limit)
+        for subdomain in subdomains_iterator:
+            attr = MISPAttribute()
+            attr.from_dict(**dict(type='domain', value=subdomain.id))
+            self.misp_event.add_attribute(**attr)
+            domain_object.add_reference(attr.uuid, 'subdomain')
```

Comment:
Refactor suggestion:
```suggestion
        for relationship_name, misp_name in [('siblings', 'sibling-of'), ('subdomains', 'subdomain')]:
            rel_iterator = self.client.iterator(f'/domains/{domain_report.id}/{relationship_name}', limit=self.limit)
            for item in rel_iterator:
                attr = MISPAttribute()
                attr.from_dict(**dict(type='domain', value=item.id))
                self.misp_event.add_attribute(**attr)
                domain_object.add_reference(attr.uuid, misp_name)
```